libs/contrib/Data/Fun/Extra.idr

```diff
@@ -8,15 +8,15 @@ import Data.HVect
 
 ||| Apply an n-ary function to an n-ary tuple of inputs
 public export
-uncurry : {ts : Vect n Type} -> {cod : Type} -> Fun ts cod -> HVect ts -> cod
-uncurry f [] = f
-uncurry f (t::ts) = uncurry (f t) ts
+uncurry : {n : Nat} -> {0 ts : Vect n Type} -> Fun ts cod -> HVect ts -> cod
```

Comment:
Now that I think about it, the `n` can *also* be `0` here because the
`HVect` argument gives you enough information! In `curry` however you
really cannot get away with not having it.

Reply:
Yes, I noticed it elsewhere, but missed it here. Thanks!